Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIPermissionsCocoa.mm

```diff
@@ -107,6 +113,18 @@
     HashSet<String> permissions, origins;
     parseDetailsDictionary(details, permissions, origins);
 
+    NSString *errorMessage;
+    WebExtension::MatchPatternSet matchPatterns;
+    if (!validatePermissionsDetails(permissions, origins, matchPatterns, @"permissions.remove()", &errorMessage)) {
+        callback->reportError(errorMessage);
```

Comment:
Should add the comment: `// Chrome reports this error as callback error and not an exception, so do the same.`